Verbose test results

PhilVargas opened the pull request:
Add a `--verbose` config flag to pretty print the test suite individually
Relates to issue #4 and takes into account some of the suggestions from pull #92 
contributors include myself @PhilVargas and @arcin

called using `jest --verbose`

example of when the test suite passes:
![pass image](http://i.imgur.com/CLLyYCq.png)
example of when the test suite fails:
![fail image](http://i.imgur.com/Ujnq6wc.png)

DmitrySoshnikov commented:
This looks really good. We can probably change recursion (in building and reporting the tree) with a loop for a better performance, although, if it's not noticeable performance win, we can keep recursion since it's cleaner.

For the visual output, you may want consider instead of coloring the whole title as green, to use just a green checkmark (or red cross in case of failure) -- you can check how Mocha framework displays it.

PhilVargas commented:
Thanks for the feedback @DmitrySoshnikov ! Went through and linted all the files, moved the `_formatMsg` function into utils so it can be exported out to the various reporters, and changed the style of the verbose output to more resemble mocha instead of rspec. Screenshots of the updated output are below:
failing test
![failing test](http://i.imgur.com/OwjO3lH.png)

passing test
![passing test](http://i.imgur.com/k2NNnWd.png)

DmitrySoshnikov commented:
OK, this looks perfect, thanks for the contribution!

(worth also measuring the time with verbose and standard outputs for analysis)
